### PR TITLE
setup Ocean typography

### LINKED ISSUE #116: Wrong typography
The current app visible under https://df.oceandao.org uses [IBM's corporate branding typeface](https://www.ibm.com/plex/) for whatever reason. Besides the fact we really don't want to be associated with that company, this app should follow [Ocean's brand guidelines](https://oceanprotocol.com/art#styleguide).

Screen Shot 2022-06-16 at 16 14 40

As this app is created by members of the oceanprotocol org, you can access this:
https://github.com/oceanprotocol/typographies

Instructions are in the above readme, increases app size a bit but you don't have to move over any font files, they are all delivered with `@import` in css so no licensing issues by making this repo here public with a copyrighted font.

### PR DESCRIPTION
**Pull request by kremalicious**
* closes #116

### CODE REVIEW

**Comment by kremalicious** [comment]
@KatunaNorbert feel free to pick up from here. base64-inlined font files are loaded in app, copied over the CSS custom properties we use in market, and its global generic styles. This should give you out-of the box Ocean styling and even allows you to add light/dark mode easily by toggling a `dark` class on the `body`.

But I see almost every component overwrites many things. Also in my first 5 min. with Svelte could not figured out where the heck this IBM typeface is loaded from.

**Comment by KatunaNorbert** [comment]
Hey @kremalicious, thanks for looking into app and opening this PR, great to see your feedback and contribution!

IBM typeface seems to be loaded by importing the `carbon-components-svelte` library, IBM plex is there default typeface and Svelte seems to have been peeking that up [see docs](https://carbondesignsystem.com/guidelines/typography/overview/)

I'm going to override this with Ocean styling.

**Comment by kremalicious** [comment]
So if you picked an existing design system, then you would need to run with it, and theme it however that system wants to be themed. Pretty sure there are then some typography overrides. Cause putting `!important` on everything bis not the solution, as that would require you to add font declarations on pretty much each and every element.

Debatable if 2 screens with 2 tables need a full design system, loading many custom fonts and all. But if it's in place you have to use it and work with its features. If we start overwriting everything coming from the design system, then what's the point of using it in the first place?

**Comment by KatunaNorbert** [comment]
I agree, we shouldn't have 2 custom fonts and one overwriting the other, we should ideally only have the Ocean typography.
Was looking into importing only the needed components from `carbon-components-svelte` but didn't find a quick answer so I put it on hold right now. Also completely removing imported components from `carbon-components-svelte` is not an option right now because it will take some time to replace those components.

**Comment by EnzoVezzaro** [comment]
I still see this font being imported. Is there any way to get rid of it completely? 

Screenshot 2023-01-30 at 10 54 54

**Comment by KatunaNorbert** [comment]
I'm looking into it right now

**Comment by EnzoVezzaro** [comment]
> I'm looking into it right now

Great 🤙🏽 I found this in the docs. Don't know if it's helpful, but [here it's](https://github.com/carbon-design-system/carbon/blob/7be55cd74d0aaf522751e22c53ed45555f1d6964/docs/guides/ibm-plex.md#font-face).

**Comment by jamiehewitt15** [comment]
Not sure why data farming is called "Farms" here, might be because this PR is behind main. Could you do a pull from main?

image

**Comment by KatunaNorbert** [comment]
This is how it's now called on the main as well, this was changed a few weeks ago, I'm not sure what are you comparing it to, there might be some PRs in review that are not up to date with main and create the confussion

**Comment by EnzoVezzaro** [comment]
typography looks good, I just found this css issue on the table button:

Button NORMAL


Button HOVER
Screenshot 2023-02-02 at 07 52 38

**Comment by KatunaNorbert** [comment]
Good catch!

**Comment by EnzoVezzaro** [comment]
this is not really an issue with typography, but I see the tooltip on mobile is off:

Screenshot 2023-02-02 at 09 50 53

**Comment by KatunaNorbert** [comment]
Yeah looks like there was a margin bottom set to paragraphs.

File: src/components/common/Button.svelte [hunk 1]
```diff
@@ -66,18 +95,25 @@
   }
   .buttonSecondary {
     background-color: var(--brand-grey-dimmed);
-    color: var(--brand-grey-light);
     border-radius: var(--border-radius);
     border: 1px solid var(--brand-grey-lighter);
   }
   .textOnly {
     background-color: transparent;
     color: var(--brand-grey-light);
```
Comment (EnzoVezzaro) [inline src/components/common/Button.svelte:103]: duplicated rule

**Review by EnzoVezzaro — APPROVED** [review]
There's some duplicated CSS rules but other than that, typography looks fine

**Comment by KatunaNorbert** [comment]
Seems to be working, I tried to remove it from Tooltip component and had some issue. We have tooltips inside all pages so I the style will be loaded each time. That wasn't the case for the Table component.

File: src/components/common/Table.svelte [hunk 2]
```diff
@@ -354,7 +356,19 @@
...
 
-<style>
+<style lang="scss" global>
+  $css--font-face: false;
+$css--helpers: false;
+$css--body: false;
+$css--use-layer: false;
+$css--reset: false;
+$css--default-type: false;
+$css--plex: false;
```
[the remark this answers is not in the document]
Comment (KatunaNorbert) [reply src/components/common/Table.svelte]: I was trying to set them inside global style file but wasn't working for me, so I decided to just leave inside each component.

[the remark this answers is not in the document]
Comment (EnzoVezzaro) [reply src/components/common/Table.svelte]: maybe we can move forward this PR the way it is, and create an issue with low priority to work on this. That can be something interesting to work on.

**Comment by EnzoVezzaro** [comment]
Maybe I didn't notice it before, but I don't see "selected networks" on this PR.

On preview PR:
Screenshot 2023-02-07 at 07 53 01

On prod:
Screenshot 2023-02-07 at 07 54 34

**Comment by kremalicious** [comment]
fyi, all your headings are semantically actually not headings. Which is bad in itself for accessibility reasons where screen readers can't figure out the structure of any page because of it.

But also because of that, the actual heading font, "Sharp Sans Display No. 1 Bold", is not kicking in for headings. You are only using default "Sharp Sans" in its bold variant for the headings which is different than the mentioned heading font. Until style guide is changed this is 3hat should be followed https://oceanprotocol.com/art/#styleguide

**Comment by EnzoVezzaro** [comment]
the font side now looks off. It's way too big now. 

PR: 
Screenshot 2023-02-20 at 11 55 34

Main:
Screenshot 2023-02-20 at 11 55 40

**Comment by KatunaNorbert** [comment]
@EnzoVezzaro is it too big? I was following the market font sizes for different elements, especially the headings.
I made the section titles as h2 and card titles as h3, and that increased their font sizes.

**Comment by EnzoVezzaro** [comment]
> @EnzoVezzaro is it too big? Maybe before was too small. I was following the market font sizes for different elements, especially the headings.
> I made the section titles as h2 and card titles as h3, and that increased their font sizes.

Yeah, I see. Now the text looks bigger but the font-size is correct. Let approve this typography issue to focus on these little things in #466

**Review by EnzoVezzaro — APPROVED** [review]

**Comment by idiom-bytes** [comment]
1 point assigned to deploy the PR.

**Pull request merged**